semantics/src/main/kotlin/com/strumenta/kolasu/semantics/symbol/description/SymbolDescription.kt

```diff
@@ -3,46 +3,173 @@ package com.strumenta.kolasu.semantics.symbol.description
 import com.strumenta.kolasu.language.KolasuLanguage
 import com.strumenta.kolasu.model.Named
 import com.strumenta.kolasu.model.Node
+import com.strumenta.kolasu.model.ReferenceByName
+import kotlin.reflect.KClass
+import kotlin.reflect.full.allSuperclasses
 
+/**
+ * The description of a [Node] from the outside,
+ * i.e. from other trees not containing the node.
+ * @property name the name of the symbol
+ * @property identifier the identifier of the described node
+ * @property type a description of the type of the described node
+ * @property fields the visible fields of the described node
+ *
+ * @author Lorenzo Addazi <lorenzo.addazi@strumenta.com>
+ **/
 data class SymbolDescription(
-    override val name: String,
-    val identifier: String,
-    val types: List<String>,
+    override var name: String,
+    var identifier: String,
+    val type: TypeDescription,
     val fields: Map<String, ValueDescription>
 ) : Node(), Named
 
+/**
+ * The type description of a symbol.
+ * @property name the name of the type
+ * @property superTypes possible super types
+ *
+ * @author Lorenzo Addazi <lorenzo.addazi@strumenta.com>
+ **/
+data class TypeDescription(
```

Comment:
I would also make TypeDescription not nodes.
Also, for which purpose do we need TypeDescriptions?

Reply:
Type descriptions are used to describe the type of the node described by a symbol - its name and possible supertypes. These are used when reasoning over the type of symbol without loading the corresponding node. I think we could define these without extending Node, is there any issue related to these being Node instances?

Reply:
Oh, I see. Sorry, I thought this represented types calculated for expressions. I instead understand that they represent instead the type of the Node (e.g., the AST class). In Kolasu 1.6., we have classes representing the metamodel (e.g., Concept) and we could use that for this goal. 

> is there any issue related to these being Node instances?

The issue is that a Node should represent an AST Node. Using it for other things could lead to confusion

Reply:
I think keeping these as Nodes would allow us to re-use most of the Kolasu API for navigating, tracking positions, representing references. Is there a document where we decided that Kolasu nodes can only represent nodes strictly contained in the AST? I feel this is not really the current practice, e.g. nodes have also been used to represent types or external symbol representations in past discussions/projects.

Reply:
I added a comment about this above. I agree this should be something well documented and defined, while now it is a sort of grey area

Reply:
P.S. I understand on this topic things are changing (based on what came up working on the Code Insight Studio and applying some ideas in practice) and that we today had limited time to discuss all aspects of it during the technical meeting. I am sorry for any complication caused by these updates and I am happy to have a meeting to discuss these things with more calm, if it is useful